usr/share/rear/conf/Ubuntu.conf

```diff
@@ -1,2 +1,18 @@
 PROGS=( "${PROGS[@]}" initctl MAKEDEV )
 
+#####
+# TCG Opal 2 PBA system only: Include Plymouth graphical boot animation
+
```

Comment:
I am not a TCG Opal 2 PBA user but from plain looking at th code
I wonder if including Plymouth graphical boot animation is really Ubuntu-only specific
because it happens in a Ubuntu-only config file usr/share/rear/conf/Ubuntu.conf
I would expect including Plymouth graphical boot animation
happens in a general TCG Opal 2 PBA specific script?

Reply:
You are right, Plymouth is used by other distros like Fedora, too and my initial intention was to support it everywhere. However, Plymouth integration seems so distro-specific that I had decided not to get other users into trouble with (on their distro) untested code.

One example: While Plymouth can fall back to a text mode boot splash, it will do a graphical boot splash if it can. It then needs the right font file(s) to display the password prompt (otherwise each character will appear as a placeholder rectangle). So which one is the right font on your distro? You cannot decipher from Plymouth installation directly as Plymouth has its own programming language and a distro-specific script written in that language decides about the font. On Ubuntu, including 4 MiB of font files from `/usr/share/fonts/truetype/ubuntu` is sufficient. Other distros might use other font files, maybe not even TrueType ones. Pulling the entire contents of `/usr/share/fonts` on my system would mean adding 120 MiB to a PBA that's normally only 40 MiB in size. In the end, we'd exceed the maximum allowed PBA size of 128 MiB.

We also can't be sure where each distro decides to put all of its Plymouth components. On Ubuntu: Why are some shared libraries sitting under `/usr/lib/x86_64-linux-gnu/plymouth` along with non-library files, while others reside under names like `/lib/x86_64-linux-gnu/libply-*`?

So I think the current way is safer, the PBA will still work without Plymouth due to its own text-mode fallback, just not look as nice.

Reply:
@OliverO2 
thank you for the explanation.
Now the Ubuntu specific config file is perfectly fine with me.
It can be used as an example for users on other distros.